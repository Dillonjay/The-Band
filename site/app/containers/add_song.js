import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { addSong } from '../actions/song_actions';


class AddSong extends React.Component {
	render() {
		return (
			<div>
				<h2>{ this.props.name }</h2>
				<form onSubmit={(e) => {
					e.preventDefault()
					this.props.addSong({title: "saaahdude", artist:"dillon", genre:"rap"})
				}}>
					<input></input>
					<button type="submit">Add</button>
				</form>
			</div>
		)
	}
};

const { string, func, isRequired } = React.PropTypes;

AddSong.propTypes = {
	name: string,
	addSong: func.isRequired
}

AddSong.defaultProps = {
	name: 'Dillon'
}

function mapStateToProps(state) {
	return { 
		songs : state.songs 
	}; 
};

function matchDispatchToProps(dispatch) {
	return bindActionCreators({ addSong: addSong }, dispatch)
}
export default connect(mapStateToProps, matchDispatchToProps)(AddSong);

