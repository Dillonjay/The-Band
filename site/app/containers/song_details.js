import React from 'react';
import { connect } from 'react-redux';


class SongDetails extends React.Component {
    render() {
        if(!this.props.song) {
            return (
                <div>No SOng in the props</div>
            )
        }
        const { title, artist, genre } = this.props.song;
        return (
            <div>
                <h1>{ title }</h1>  
                <h2>{ artist }</h2>
                <h3>{ genre }</h3>
            </div>
        )
    }
}

const { shape, string } = React.PropTypes;

SongDetails.propTypes = {
    song: shape({
        title: string,
        artist: string,
        genre: string
    })
}

function mapStateToProps(state) {
    return { 
        song: state.activeSong
    }; 
}

// Component not wrapped in redux for testing purposes. 
export const Unwrapped = SongDetails;

export default connect(mapStateToProps)(SongDetails);