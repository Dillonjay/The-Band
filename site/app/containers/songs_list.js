import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { selectSong } from '../actions/index.js';

class SongList extends React.Component {
	render() {
		return (
			<ul>
				{this.showSongList()}
			</ul>
		)
	}
	showSongList(songs) {
		return this.props.songs.map((song,i) => {
			return (
					<li 
					key={i}
					onClick={() => this.props.selectSong(song)}
					>
						{song.title}
					</li>
				)
		})
	}
};

function mapStateToProps(state) {
	return { 
		songs : state.songs 
	}; 
};

function matchDispatchToProps(dispatch) {
	return bindActionCreators({ selectSong: selectSong }, dispatch)
}

export default connect(mapStateToProps, matchDispatchToProps)(SongList);