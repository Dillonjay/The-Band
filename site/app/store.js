/* global window:false */
import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducers/index';

const middleware = applyMiddleware(thunk);
const store = createStore(reducers, compose(
    middleware,
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : (f) => f
)); 

export default store;
