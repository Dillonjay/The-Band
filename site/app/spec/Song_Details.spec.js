import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../store';
import { Unwrapped as UnwrappedSongDetails } from '../containers/song_details';

test('SongDetails Snapshot Test', () => {
	const component = shallow(<UnwrappedSongDetails song={ {title: 'Bird', artist: 'Anderson Paak', genre: 'hiphop'} } /> );
	const tree = shallowToJson(component);
	expect(tree).toMatchSnapshot();
})
