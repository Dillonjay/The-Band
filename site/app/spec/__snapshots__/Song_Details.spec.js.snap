exports[`test SongDetails Snapshot Test With Song 1`] = `
<div>
  <h1>
    Bird
  </h1>
  <h2>
    Anderson Paak
  </h2>
  <h3>
    hiphop
  </h3>
</div>
`;
